src/clib/pio_file.c

```diff
@@ -405,9 +417,9 @@ int PIOc_sync (int ncid)
   msg = PIO_MSG_SYNC;
 
   if(ios->async_interface && ! ios->ioproc){
-    if(ios->compmaster) 
+    if(!ios->comp_rank) 
```

Comment:
In the MPI C API, compmaster is not a boolean, it is an integer, set to either 0 or MPI_ROOT.

I have changed the if statement as you suggest.